app/src/scenes/phase1/changeSejour.js

```diff
@@ -0,0 +1,481 @@
+
+import { DropdownItem, DropdownMenu, DropdownToggle, UncontrolledDropdown } from "reactstrap";
+
+export default function changeSejour() {
+  const young = useSelector((state) => state.Auth.young);
+  const [newSejour, setNewSejour] = useState("");
+  const [motif, setMotif] = useState("");
+  const [modalConfirmControlOk, setmodalConfirmControlOk] = useState(false);
+  const [modalConfirmGoalReached, setmodalConfirmGoalReached] = useState(false);
+  const [sejours, setSejours] = useState(null);
+  const [isEligible, setIsElegible] = useState(false);
+  const [sejourGoal, setSejourGoal] = useState(null);
+  const [messageTextArea, setMessageTextArea] = useState("");
+
+  const motifs = [
+    "Non disponibilité pour motif familial ou personnel",
+    "Non disponibilité pour motif scolaire ou professionnel",
+    "L'affectation ne convient pas",
+    "Impossibilité de se rendre au point de rassemblement",
+    "Autre",
+  ];
+
+  useEffect(() => {
+    (async function getInfo() {
+      try {
+        const { ok, code, data } = await api.post("/cohort-session/eligibility/2022", {
+          birthDate: young.birthdateAt,
+          schoolLevel: young.schoolLevel,
+          department: young.department,
+        });
+        const sejourGoal = data.map((e) => {
+          // les dates de fin d'inscription aux séjours ne sont pas renseignés pour le moment
+          //var date = new Date();
+          //console.log(date.toISOString());
+          //if (e.inscriptionLimitDate > date.toISOSString())    date de fin de d'inscription aux séjours à récupérer
+          return { sejour: e.id, goal: e.goalReached };
+        });
+
+        const sejour = sejourGoal.map((e) => e.sejour);
+
+        setSejours(sejour);
+        setIsElegible(!!data);
+        setSejourGoal(sejourGoal);
+      } catch (e) {
+        toastr.error("Oups, une erreur est survenue", translate(e.code));
+      }
+    })();
+  }, []);
+
+  const onConfirmer = () => {
+    console.log("raison : " + messageTextArea);
+    if (newSejour && motif) {
+      var isGoalTrue = sejourGoal.filter((obj) => {
+        if (obj.goal === true && obj.goal === newSejour) return obj.sejour === newSejour;
+      });
+      //si le volontaire est en statut de phase 1 “affectée” et que les objectifs de recrutement sont atteint pour le nouveau séjour choisi
+
+      if (isGoalTrue.length === 0) {
+        setmodalConfirmControlOk(true);
+      } else {
+        setmodalConfirmGoalReached(true);
+      }
+    } else {
+      toastr.info("Veuillez renseigner tous les champs");
+    }
+  };
+
+  const handleChangeSejour = async () => {
+    try {
+      await api.put("/young/" + young._id + "/change-cohort/" + newSejour, { cohortChangeReason: motif });
+      await api.put("/young/" + young._id + "/change-cohort/" + newSejour, { cohortDetailedChangeReason: messageTextArea });
+      await api.put("/young/" + young._id + "/change-cohort/" + newSejour, { cohort: newSejour });
```

Comment:
on peut le faire en un seul call (avec la nouvelle route proposée plus haute)
(1 seul call api, a seul call a la db)
```suggestion
      await api.put("/young/" + young._id + "/change-cohort", { cohortChangeReason: motif, cohortDetailedChangeReason: messageTextArea, cohort: newSejour });
```